common/src/test/java/org/apache/falcon/entity/FileSystemStorageTest.java

```diff
@@ -440,6 +441,46 @@ public void testListing(String availabilityFlag, Frequency frequency, TimeZone t
         }
     }
 
+    @Test (dataProvider = "testListingDataProvider")
+    public void testInstanceListing(String availabilityFlag, Frequency frequency, TimeZone timeZone,
+                                    Date start, Date end) throws Exception {
+        EmbeddedCluster firstCluster = EmbeddedCluster.newCluster("TestFeedListing", false);
+        FileSystem fs = firstCluster.getFileSystem();
+        ConfigurationStore.get().publish(EntityType.CLUSTER, firstCluster.getCluster());
+
+
+        EmbeddedCluster secondCluster = EmbeddedCluster.newCluster("TestFeedInstanceListing", false);
+        ConfigurationStore.get().publish(EntityType.CLUSTER, secondCluster.getCluster());
+
+        try {
+            Feed feed = getFeed(availabilityFlag, frequency, timeZone);
+            Cluster cluster = new Cluster();
+            cluster.setName("TestFeedInstanceListing");
+            feed.getClusters().getClusters().add(cluster);
+            Validity validity = new Validity();
```

Comment:
We should be testing for feed with two clusters.Please make that change so that your code is thoroughly tested.